frontends/ast/genrtlil.cc

```diff
@@ -1090,7 +1090,6 @@ void AstNode::detectSignWidthWorker(int &width_hint, bool &sign_hint, bool *foun
 		}
 		if (str == "\\$size" || str == "\\$bits" || str == "\\$high" || str == "\\$low" || str == "\\$left" || str == "\\$right") {
 			width_hint = 32;
```

Comment:
Thank you for fixing this! I think this should also use `max` to avoid trimming the width of larger expressions. I wouldn't mind if that is fixed separately, if you'd prefer.